### PR TITLE
Minor changes to CovidModel config parameters and updated report

### PR DESCRIPTION
**Pull request by Shruthi42**
* Do inference on both validation and test sets by default in the `CovidModel` config
* Add parameter `pretraining_dataset_id` to `NIH_COVID_BYOL` to specify the training dataset
* Update report in `CovidModel`

### CODE REVIEW

File: InnerEye/ML/configs/classification/CovidModel.py [hunk 1]
```diff
@@ -237,45 +239,118 @@ def compute_and_log_metrics(self,
...
-                predictions.append(target_df[LoggingColumns.ModelOutput.value])
-                labels.append(target_df[LoggingColumns.Label.value])
+                predictions.append(target_df[output_prefix].item())
+                labels.append(target_df[label_prefix].item())
 
-            return pd.DataFrame.from_dict({LoggingColumns.Patient.value: [df.iloc[0][LoggingColumns.Patient.value]],
-                                           LoggingColumns.ModelOutput.value: [np.argmax(predictions)],
-                                           LoggingColumns.Label.value: [np.argmax(labels)]})
+            pred_cvx03vs12 = predictions[1] + predictions[2]
+            label_cvx03vs12 = 1 if labels[1] or labels[2] else 0
```
Comment (dccastro) [inline InnerEye/ML/configs/classification/CovidModel.py:269]: minor: for this kind of statement I believe the intent would be clearer with `int(labels[1] or labels[2])`

File: InnerEye/ML/configs/classification/CovidModel.py [hunk 2]
```diff
@@ -237,45 +239,118 @@ def compute_and_log_metrics(self,
...
-        msg += f"Multiclass Accuracy Test: {get_accuracy(pd.read_csv(test_metrics))}\n" if test_metrics.exists() else ""
+            return pd.DataFrame.from_dict({LoggingColumns.Patient.value: [df.iloc[0][LoggingColumns.Patient.value]],
+                                           output_prefix: [np.argmax(predictions)],
+                                           label_prefix: [np.argmax(labels)],
+                                           f"{output_prefix}_CVX03vs12": pred_cvx03vs12,
+                                           f"{label_prefix}_CVX03vs12": label_cvx03vs12,
+                                           f"{output_prefix}_CVX0vs3": pred_cvx0vs3,
+                                           f"{label_prefix}_CVX0vs3": label_cvx0vs3,
+                                           f"{output_prefix}_CVX1vs2": pred_cvx1vs2,
+                                           f"{label_prefix}_CVX1vs2": label_cvx1vs2})
```
Comment (dccastro) [inline InnerEye/ML/configs/classification/CovidModel.py:291]: minor suggestion: as these keys/column names are used multiple times, would it be helpful to define them as variables at the top, e.g. `output_03vs12_col` etc.?

**Review by dccastro — CHANGES_REQUESTED** [review]

File: InnerEye/ML/configs/classification/CovidModel.py (hunk 1 above)
Comment (dccastro) [reply InnerEye/ML/configs/classification/CovidModel.py]: sorry, I meant `label_cvx03vs12 = int(labels[1] or labels[2])`. on second thought it may be better to revert to how it was initially, for consistency with the rest 🙂

**Review by dumbledad — APPROVED** [review]

**Review by dccastro — APPROVED** [review]
